internal/graph/memory.go

```diff
@@ -31,6 +31,7 @@ import (
 // Memory is a memory based PredecessorFinder.
 type Memory struct {
 	predecessors sync.Map // map[descriptor.Descriptor]map[descriptor.Descriptor]ocispec.Descriptor
+	successors   sync.Map // map[descriptor.Descriptor]map[descriptor.Descriptor]ocispec.Descriptor
```

Comment:
Since the successor list of a node is immutable and we don't need random access on the successors, the mapping can be from a descriptor to a successor list (`map[descriptor.Descriptor][]ocispec.Descriptor`).

Reply:
changed both `successors` and `predecessors` to the regular map, and added locks in `index` and `Remove`.